Automattic/simplenote-macos
# Magic Links: UI Updates Mk1

### Fix
This is the first step in reshaping the SNMac authentication to meet the new designs using magic links as the primary mode of authentication.  

| signup view | login with email |
|--------|--------|
| <img src="https://cdn-std.droplr.net/files/acc_593859/6DyFm0"/> | <img src="https://cdn-std.droplr.net/files/acc_593859/yZj3qF" /> | 

In this PR I have done the following:
- Removed the switch mode button/view/actions
- changed the Sign in view -> login view button to use the secondary button as it's trigger
- refactored the ui for the sign in with WP button to match the design on iOS

What is not done here:
- solve the sizing problem when hitting the back button on the navigation controller
- Adding the next view where you can code auth or password auth
- refactor authentication mode to be more generic
These changes will be coming in the next PR

### Test
1. Launch SNMac, tap on the Login button and confirm the login with email view appears
2. click on the login with Wordpress button and confirm that still works
3. tap on the continue with password button and confirm the password appears
4. Attempt login with email, confirm that works
5.  attempt login with password, confirm that works

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> These changes do not require release notes.
